test/unit/bone.test.js

```diff
@@ -5,119 +5,178 @@ const { Bone, DataTypes, connect } = require('../..');
+      ]);
+      assert.ok(User.timestamps);
+      assert.equal(User.timestamps.createdAt, 'createdAt');
+      assert.equal(User.timestamps.updatedAt, 'updatedAt');
+      assert.equal(User.timestamps.deletedAt, 'deletedAt');
+    });
+
+    it('should mark primaryKey', async function() {
+      class User extends Bone {
+        static attributes = {
+          ssn: { type: STRING, primaryKey: true },
+        }
+      }
+      User.load([
+        { columnName: 'ssn', columnType: 'varchar', dataType: 'varchar', primaryKey: true },
+      ]);
+      assert.equal(User.primaryKey, 'ssn');
+      assert.equal(User.primaryColumn, 'ssn');
+    });
   });
 
-  it('should mark timestamps', async function() {
-    class User extends Bone {
-      static attributes = {
-        createdAt: DATE,
-        updatedAt: DATE,
-        deletedAt: DATE,
+  describe('=> Bone.loadAttribute()', function() {
+    it('should make the loaded attribute enumerable', async function() {
+      class User extends Bone {}
+      User.loadAttribute('foo');
+      assert.ok(Object.getOwnPropertyDescriptor(User.prototype, 'foo').enumerable);
+    });
+
+    it('should make sure both getter and setter exist', async function() {
+      class User extends Bone {
+        get foo() {
+          return this.attribute('foo');
+        }
       }
-    }
-    User.load([
-      { columnName: 'id', columnType: 'bigint', dataType: 'bigint', primaryKey: true },
-      { columnName: 'created_at', columnType: 'timestamp', dataType: 'timestamp' },
-      { columnName: 'updated_at', columnType: 'timestamp', dataType: 'timestamp' },
-      { columnName: 'deleted_at', columnType: 'timestamp', dataType: 'timestamp' },
-    ]);
-    assert.ok(User.timestamps);
-    assert.equal(User.timestamps.createdAt, 'createdAt');
-    assert.equal(User.timestamps.updatedAt, 'updatedAt');
-    assert.equal(User.timestamps.deletedAt, 'deletedAt');
+      User.loadAttribute('foo');
+      assert.ok(Object.getOwnPropertyDescriptor(User.prototype, 'foo').enumerable);
+      assert.ok(Object.getOwnPropertyDescriptor(User.prototype, 'foo').set);
+    });
   });
 
-  it('should mark primaryKey', async function() {
-    class User extends Bone {
-      static attributes = {
-        ssn: { type: STRING, primaryKey: true },
+  describe('=> Bone.renameAttribute()', function() {
+    it('should make the renamed attribute enumerable', async function() {
+      class User extends Bone {
+        static attributes = {
+          foo: { type: STRING },
+        }
       }
-    }
-    User.load([
-      { columnName: 'ssn', columnType: 'varchar', dataType: 'varchar', primaryKey: true },
-    ]);
-    assert.equal(User.primaryKey, 'ssn');
-    assert.equal(User.primaryColumn, 'ssn');
+      User.load([
+        { columnName: 'foo', columnType: 'varchar', dataType: 'varchar' },
+      ]);
+      User.renameAttribute('foo', 'bar');
+      assert.ok(Object.getOwnPropertyDescriptor(User.prototype, 'bar').enumerable);
+    });
```

Comment:
It's expected that attributes should always present in `toJSON()` or `toObject()` results. Make renamed attribute enumerable as well.